src/librustc_mir/borrow_check/nll/region_infer/error_reporting/region_name.rs

```diff
@@ -681,10 +681,13 @@ impl<'tcx> RegionInferenceContext<'tcx> {
 
         let (return_span, mir_description) = match tcx.hir().get(mir_node_id) {
             hir::Node::Expr(hir::Expr {
-                node: hir::ExprKind::Closure(_, _, _, span, gen_move),
+                node: hir::ExprKind::Closure(_, return_ty, _, span, gen_move),
                 ..
             }) => (
-                tcx.sess.source_map().end_point(*span),
+                match return_ty.output {
+                    hir::FunctionRetTy::DefaultReturn(_) => tcx.sess.source_map().end_point(*span),
+                    hir::FunctionRetTy::Return(_) => return_ty.output.span(),
+                },
```

Comment:
Isn't `return_ty.output.span()` always correct?

Reply:
It isn't, I tried that first. This would point to the character after the last `|` in the closure instead. For example:

```diff
2         --> $DIR/issue-53040.rs:5:8
3          |
4       LL |     || &mut v;
-          |      - ^^^^^^ returns a reference to a captured variable which escapes the closure body
-          |      |
-          |      inferred to be a `FnMut` closure
+          |        -^^^^^
+          |        |
+          |        returns a reference to a captured variable which escapes the closure body
+          |        inferred to be a `FnMut` closure
8          |
9          = note: `FnMut` closures only have access to their captured variables while they are executing...
10         = note: ...therefore, they cannot allow references to captured variables to escape
```